src/main/java/nextstep/subway/application/SectionService.java

```diff
@@ -0,0 +1,58 @@
+package nextstep.subway.application;
+
+import lombok.RequiredArgsConstructor;
+import nextstep.subway.common.exception.CustomException;
+import nextstep.subway.common.exception.ResponseCode;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.SectionRepository;
+import nextstep.subway.domain.Sections;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.StationRepository;
+import nextstep.subway.ui.dto.section.CreateSectionRequest;
+import nextstep.subway.ui.dto.section.SectionResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+@RequiredArgsConstructor
+@Service
+public class SectionService {
+    private final SectionRepository sectionRepository;
+    private final StationRepository stationRepository;
+    private final LineRepository lineRepository;
+
+    public SectionResponse createSection(long lineId, CreateSectionRequest request) {
+        Line line = findLine(lineId);
+        Station upStation = findStation(request.getUpStationId());
+        Station downStation = findStation(request.getDownStationId());
+
+        Section section = Section.builder()
+                                 .upStation(upStation)
+                                 .downStation(downStation)
+                                 .distance(request.getDistance()).build();
+        line.addSection(section);
+        sectionRepository.save(section);
```

Comment:
아래 구문을 삭제해도 동작합니다. 왜그럴까요? 🤔 

```suggestion
//        sectionRepository.save(section);
```

Reply:
Cascade.ALL 옵션으로 부모 엔티티가 영속화 될때 자식 엔티티까지 함께 영속화 되기 때문인것 같습니다!
4f5e9ed9